chore(native filters): Expandable filter config modal

justinpark opened the pull request:
### SUMMARY
The native filters modal width is unnecessarily narrow which results in a suboptimal UX—excessive scrolling—especially when either the filter names and/or datasets/column names are long.
This commit adds an option to make the  'Add and edit filters' modal wider.

<img width="1036" alt="Sales_Dashboard" src="https://github.com/apache/superset/assets/1392866/20cd78e8-7bd5-4c11-bf19-42d01a0c5bcd">


### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

After:

https://github.com/apache/superset/assets/1392866/6027a46b-aa5f-4f1d-8656-52d7c33c9928

Before:

https://github.com/apache/superset/assets/1392866/7d6311ac-2fa8-4b9b-8656-1e0dd9b67f43


### TESTING INSTRUCTIONS
Go to dashboard and choose a dataset contains a long name column name

### ADDITIONAL INFORMATION
- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

Commit:
chore(native filters): Expandable filter config modal

john-bodley commented:
@justinpark in addition to the scroll bars, is there merit in making the default modal size wider?

justinpark commented:
> in addition to the scroll bars, is there merit in making the default modal size wider?

Sure. I can adjust the default width from 50% to 70%

Commit:
increase default width to 70%

Commit:
remove icon mock

michael-s-molina commented on superset-frontend/src/dashboard/components/nativeFilters/FiltersConfigModal/FiltersConfigModal.tsx:
```diff
@@ -580,8 +601,26 @@ function FiltersConfigModal({
+          />
+        </div>
+      }
+      width="70%"
+      {...(expanded && { width: '100%', height: '100%' })}
```
Given that `StyleModalWrapper` is already receiving the `expanded` property, could you deal with `width` and `height` there and remove these two lines?

Commit:
replace width by styled component

justinpark commented:
@michael-s-molina could you review the update?

michael-s-molina reviewed (approved):
LGTM

kasiazjc commented:
Thanks for the change @justinpark! I am worried that because the icons are so similar and close to each other, people might confuse them. I think in other places we use this icon (below) and using this one would probably help. Would it be ok to swap?
<img width="430" alt="image" src="https://github.com/apache/superset/assets/36897697/e5957cdd-d33f-44a0-b4dd-e878ee93824e">

I think in general we have to bump the width of the modal in the default mode to ~880px (and resize just after it not longer fits in the screen), so that everything is more visible. Do you think you could include it in the PR or should it be included in a new one?

justinpark committed:
update the expand icon location and adjust default width to 880px

justinpark commented:
> Thanks for the change justinpark! I am worried that because the icons are so similar and close to each other, people might confuse them. I think in other places we use this icon (below) and using this one would probably help. Would it be ok to swap? <img alt="image" width="430" src="https://user-images.githubusercontent.com/36897697/252910739-e5957cdd-d33f-44a0-b4dd-e878ee93824e.png">
> 
> I think in general we have to bump the width of the modal in the default mode to ~880px (and resize just after it not longer fits in the screen), so that everything is more visible. Do you think you could include it in the PR or should it be included in a new one?

Sounds good. I updated the location of the button to the next of the footer buttons.
<img width="283" alt="Screenshot 2023-07-12 at 1 33 09 PM" src="https://github.com/apache/superset/assets/1392866/5101ce99-097a-430a-af02-a12068d67848">

I also updated the default width to 880px as well.

kasiazjc commented:
> > Thanks for the change justinpark! I am worried that because the icons are so similar and close to each other, people might confuse them. I think in other places we use this icon (below) and using this one would probably help. Would it be ok to swap? <img alt="image" width="430" src="https://user-images.githubusercontent.com/36897697/252910739-e5957cdd-d33f-44a0-b4dd-e878ee93824e.png">
> > I think in general we have to bump the width of the modal in the default mode to ~880px (and resize just after it not longer fits in the screen), so that everything is more visible. Do you think you could include it in the PR or should it be included in a new one?
> 
> Sounds good. I updated the location of the button to the next of the footer buttons. <img alt="Screenshot 2023-07-12 at 1 33 09 PM" width="283" src="https://user-images.githubusercontent.com/1392866/253094673-5101ce99-097a-430a-af02-a12068d67848.png">
> 
> I also updated the default width to 880px as well.

Amazing, thank you!
